src/components/20-molecules/image-upload/utils/imageCompressor.js

```diff
@@ -0,0 +1,259 @@
+    if (canvas.toBlob) {
+      canvas.toBlob(resolve, options.mimeType, options.quality);
+    } else {
+      resolve(toBlob(canvas.toDataURL(options.mimeType, options.quality)));
+    }
+  });
+
+/**
+ * @param {object} res
+ * @param {object} _image
+ * @param {object} _options
+ * @param {object} file
+ * @return {Promise}
+ * @function onResultCanvas
+ */
+const onResultCanvas = (res, _image, _options, file) => {
+  const image = _image;
+  const options = _options;
+  let _result = res;
+  if (URL) {
+    URL.revokeObjectURL(image.src);
+  }
+  if (_result) {
+    // Returns original file if the _result is greater than it and without size related options
+    if (
+      _result.size > file.size &&
+      !(
+        options.width > 0 ||
+        options.height > 0 ||
+        options.maxWidth < Infinity ||
+        options.maxHeight < Infinity ||
+        options.minWidth > 0 ||
+        options.minHeight > 0
+      )
+    ) {
+      _result = file;
+    } else {
+      const date = new Date();
+      _result.lastModified = date.getTime();
+      _result.lastModifiedDate = date;
+      _result.name = file.name;
+
+      // Convert the extension to match its type
+      if (_result.name && _result.type !== file.type) {
+        _result.name = _result.name.replace(
+          REGEXP_EXTENSION,
+          imageTypeToExtension(_result.type)
+        );
+      }
+    }
+  } else {
+    // Returns original file if the _result is null in some cases.
+    _result = file;
+  }
+  if (options.success) {
+    options.success(_result);
+  }
+  return Promise.resolve(_result);
+};
+
+/**
+ * @param {object} file
+ * @param {object} _options
+ * @return {Promise}
+ * @function createPromise
+ */
+const createPromise = (file, _options) => {
+  const options = _options;
+  if (!isBlob(file)) {
+    return false;
+  }
+  if (!isImageType(file.type)) {
+    return false;
+  }
+  // const image = new Image();
+  const image = document.createElement('img');
+  return new Promise((resolve, reject) => {
+    if (URL) {
+      resolve(URL.createObjectURL(file));
+    } else if (FileReader) {
+      const reader = new FileReader();
+
+      reader.onload = e => resolve(e.file.result);
+      reader.onabort = reject;
+      reader.onerror = reject;
+      reader.readAsDataURL(file);
+    } else {
+      // eslint-disable-next-line prefer-promise-reject-errors
+      reject('The current browser does not support image compression.');
+    }
+  })
+    .then(url => createImage(url, file, image))
+    .then(({ width, height }) =>
+      appendToCanvas(width, height, _options, file, image)
+    )
+    .then(res => onResultCanvas(res, image, options, file))
+    .catch(err => {
+      if (!options.error) {
+        throw err;
+      }
+
+      options.error(new Error(err));
+    });
+};
+
+/**
+ * @param {object} filesList
+ * @param {object} _options
+ * @return {Promise}
+ * @function compressedImage
+ */
```

Comment:
Why do you have documentation in this file? I would either remove it or enrich it with a description. A lot of IDE's can do autocomplete anyways and don't need a documentation that just repeats the variable names. Documentation gets outdated quickly, so we should try to avoid it as much as possible imho.

Reply:
Just copied and thought a little documentation doesn't hurt. But I agree. It doesn't add any value.

Reply:
Done

Reply:
done